source/extensions/filters/http/composite/config.h

```diff
@@ -30,23 +30,6 @@ class CompositeFilterFactory
       const envoy::extensions::filters::http::composite::v3::Composite& proto_config,
       const std::string& stats_prefix, DualInfo dual_info,
       Server::Configuration::ServerFactoryContext& context) override;
-
-  Server::Configuration::MatchingRequirementsPtr matchingRequirements() override {
```

Comment:
I don't think we should completely remove the matching requirement, instead can we just expand it with the matching type you want?

Matching on response path could be problematic. You can find a more detailed example [here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api).
"""
An example of this would be to have a matching tree that attempts to skip the gRPC-Web filter based on response headers: clients assume that if they send a gRPC-Web request to Envoy, the filter will transform that into a gRPC request before proxying it upstream, then back into a gRPC-Web response on the encoding path. By skipping the filter based on response headers, the forward transformation will happen (the upstream receives a gRPC request), but the response is never converted back to gRPC-Web. As a result, the client will receive an invalid response back from Envoy. If the skip action was instead resolved on trailers, the same gRPC-Web filter would consume all the data but never write it back out (as this happens when it sees the trailers), resulting in a gRPC-Web response with an empty body.
"""

Reply:
The problem is that my type is private, it's not part of Envoy OSS. I don't think it makes sense to add a private type (e.g. `types.example.com/my.new.input.matcher`) to the open source repo.

It sounds like the concern here is that certain parts of the HttpRequestData are not available at the time the matching tree is executing as we have already documented in .https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api.

It does sound like a problem, but I don't think the right solution is to enforce this via an allowlist hard-coded in the Envoy codebase.

One possibility is to split HttpRequestData into separate types that better reflect the phases of the request lifecycle (i.e. split into HttpRequestData and HttpResponseData) to be matched upon.

Reply:
I can see it will be problematic to private type, but my main point is that this restriction/requirement here is necessary. 

As more use cases coming out like here, it probably better to be implemented in a more dynamic way (like c++ subclass query) , but that is just implementation details that doesn't conflict with my original point.

Reply:
Agreed we should fix this, but this restriction is currently preventing users from implementing their own custom logic, regardless of whether they would run into the scenario put forth in https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api.

I'd rather not have to solve this deeper issue to use this extension point. 

Is there a documentation improvement I can make to warn users of this possibility when using the Composite filter?